### PR TITLE
Migrate router and registry startup functions from util.sh

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@Miciah PTAL the last commit

### CODE REVIEW

File: hack/lib/start.sh [hunk 1]
```diff
@@ -596,23 +596,38 @@ function os::start::internal::print_server_info() {
...
-function install_router() {
+# os::start::router installs the OpenShift router and optionally creates
+# the server cert as well.
+#
+# Globals:
+#  - CREATE_ROUTER_CERT
+#  - MASTER_CONFIG_DIR
+#  - API_HOST
+#  - ADMIN_KUBECONFIG
+#  - USE_IMAGES
```
Comment (Miciah) [inline hack/lib/start.sh]: The list of globals is missing `DROP_SYN_DURING_RESTART`.

File: hack/lib/start.sh [hunk 2]
```diff
@@ -596,23 +596,38 @@ function os::start::internal::print_server_info() {
...
+# Arguments:
+#  None
+# Returns:
+#  None
+function os::start::router() {
 	os::log::info "Installing the router"
-	oadm policy add-scc-to-user privileged -z router --config="${ADMIN_KUBECONFIG}"
+	oadm policy add-scc-to-user privileged --serviceaccount router --config="${ADMIN_KUBECONFIG}"
```
Comment (Miciah) [inline hack/lib/start.sh]: This works, but should we use `--serviceaccount=router` for consistent style?

**Comment by stevekuznetsov** [comment]
@Miciah addressed the comments
@danmcp I think this is ready for merge

**Comment by danmcp** [comment]
[merge]

**Comment by stevekuznetsov** [comment]
@danmcp the [conformance tests](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/8537/console) seem to have been cut off mid-run and got: 
```
==> openshiftdev: Downloading logs
SSH authentication failed! This is typically caused by the public/private
keypair for the SSH user not being properly set on the guest VM. Please
verify that the guest VM is setup with the proper public key, and that
the private key path for Vagrant is setup properly as well.
Build step 'Execute shell' marked build as failure
```

Thoughts? Please re-tag.

**Comment by stevekuznetsov** [comment]
Oh, they ran for five hours. Looks like we found ourselves another structural conformance flake that hits timeouts

**Pull request merged**